tests/components/sleepiq/conftest.py

```diff
@@ -1,75 +1,51 @@
-"""Common fixtures for sleepiq tests."""
-import json
+"""Common methods for SleepIQ."""
+import re
 from unittest.mock import patch
 
+from aioresponses import aioresponses
 import pytest
-from sleepyq import Bed, FamilyStatus, Sleeper
 
-from homeassistant.components.sleepiq.const import DOMAIN
+from homeassistant.components.sleepiq import DOMAIN
 from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.setup import async_setup_component
 
 from tests.common import MockConfigEntry, load_fixture
 
 
-def mock_beds(account_type):
-    """Mock sleepnumber bed data."""
-    return [
-        Bed(bed)
-        for bed in json.loads(load_fixture(f"bed{account_type}.json", "sleepiq"))[
-            "beds"
-        ]
-    ]
-
-
-def mock_sleepers():
-    """Mock sleeper data."""
-    return [
-        Sleeper(sleeper)
-        for sleeper in json.loads(load_fixture("sleeper.json", "sleepiq"))["sleepers"]
-    ]
-
-
-def mock_bed_family_status(account_type):
-    """Mock family status data."""
-    return [
-        FamilyStatus(status)
-        for status in json.loads(
-            load_fixture(f"familystatus{account_type}.json", "sleepiq")
-        )["beds"]
-    ]
-
-
 @pytest.fixture
-def config_data():
-    """Provide configuration data for tests."""
-    return {
-        CONF_USERNAME: "username",
-        CONF_PASSWORD: "password",
-    }
-
-
-@pytest.fixture
-def config_entry(config_data):
-    """Create a mock config entry."""
-    return MockConfigEntry(
+def mock_aioresponse():
+    """Mock responses for SleepIQ."""
+    BASE_URL = "https://prod-api.sleepiq.sleepnumber.com/rest/"
+    with aioresponses() as m:
+        m.put(BASE_URL + "login", body=load_fixture("sleepiq/login.json"))
+        m.get(BASE_URL + "bed?_k=0987", body=load_fixture("sleepiq/bed.json"))
+        m.get(BASE_URL + "sleeper?_k=0987", body=load_fixture("sleepiq/sleeper.json"))
+        m.get(
+            BASE_URL + "bed/familyStatus?_k=0987",
+            body=load_fixture("sleepiq/familystatus.json"),
+        )
+        m.get(re.compile(BASE_URL + ".*/foundation/.*"), status=404, repeat=True)
+        m.get(re.compile(BASE_URL + ".*/pauseMode"), payload={"pauseMode": "off"})
+        m.put(re.compile(BASE_URL + ".*/pauseMode"), repeat=True)
+
+        yield m
+
```

Comment:
Move the library instead of the http responses directly. We want to hide the abstraction from Home Assistant as much as possible.  It should only know about what the library presents and now how the underlying data is fetched.

Reply:
These are the files from the existing tests - it's no problem to change it to mock the library calls instead, but I was trying to keep it as similar as possible